### ISSUE TITLE
Problems with `filter_indices` initialization and access

### ISSUE THREAD

**Issue by chris-langfield** [ComputationalCryoEM/ASPIRE-Python#549]
From what I can tell, there are two bugs here:

#### `filter_indices` of length 1
When `ImageSource.filter_indices` is set to be a list of length 1, the `filter_indices` getter returns an object which is not subscriptable, causing errors when other methods attempt to access `filter_indices` elements. This is because of the behavior of `np.squeeze()` on an array with one element.

`ImageSource.filter_indices` getter:

```python
@property
 def filter_indices(self):
        return self.get_metadata("__filter_indices")
```

`ImageSource.get_metadata()`
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/3223d6c2c308af66f321ca2b0ddf3933045f904b/src/aspire/source/image.py#L287-L327

Script to reproduce

```
>>> from aspire.source import RelionSource
>>> relion_src = RelionSource("tests/saved_test_data/sample_relion_data.star", max_rows=1)
2022-01-26 09:24:41,000 INFO Parsing star file at: tests/saved_test_data/sample_relion_data.star
2022-01-26 09:24:41,017 INFO Created StarFile with blocks: model_class_1
2022-01-26 09:24:41,044 INFO Creating RelionSource with 1 images.
>>> relion_src.filter_indices
array(0)
>>> relion_src.filter_indices[0]
2022-01-26 09:24:52,138 CRITICAL too many indices for array: array is 0-dimensional, but 1 were indexed
```

#### `Simulation.filter_indices` are not properly initialized when no unique filters are initialized

Unless `unique_filters` is not `None`, `Simulation.filter_indices` is not initialized

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/3223d6c2c308af66f321ca2b0ddf3933045f904b/src/aspire/source/simulation.py#L85-L89

We only have one test where the `Simulation` is initialized with default unique filters: `test_simulation.py -> SingleSimTestCase.testImage()`, but this bug is not hit in the test because it doesn't directly ask for `filter_indices`

This bug is addressed by #537 by setting `filter_indices` to `np.zeros(n)` by default.

Script to reproduce

```
>>> from aspire.source import Simulation
>>> sim = Simulation(n=10, L=64)
2022-01-26 09:35:24,201 INFO Creating Simulation with 10 images.
>>> sim.filter_indices
2022-01-26 09:35:29,203 CRITICAL Missing columns and no default value provided
Traceback:
  File "<stdin>", line 1, in <module>
  File "/Users/langfield/ASPIRE-Python/src/aspire/source/image.py", line 149, in filter_indices
    return self.get_metadata("__filter_indices")
  File "/Users/langfield/ASPIRE-Python/src/aspire/source/image.py", line 332, in get_metadata
    raise RuntimeError("Missing columns and no default value provided")

```

**Comment by janden** [comment]
Good. `np.squeeze` is prone to these bugs when no axis is specified. It might be a good idea to grep after other occurrences and see if there are easy fixes.

**Comment by chris-langfield** [comment]
Closed by #537 with the `np.atleast_1d` fix
